src/MinimalDelegation.sol

```diff
@@ -108,47 +142,34 @@ contract MinimalDelegation is IERC7821, IKeyManagement, ERC1271, EIP712 {
         revert("Not implemented");
     }
 
-    // We currently only support calls initiated by the contract itself which means there are no checks needed on the target contract.
-    // In the future, other keys can make calls according to their key permissions and those checks will need to be added.
-    function _execute(bytes32 mode, Call[] memory calls) private {
-        bool shouldRevert = mode.shouldRevert();
-        for (uint256 i = 0; i < calls.length; i++) {
-            (bool success, bytes memory output) = _execute(calls[i]);
-            // Reverts with the first call that is unsuccessful if the EXEC_TYPE is set to force a revert.
-            if (!success && shouldRevert) revert IERC7821.CallFailed(output);
-        }
-    }
-
-    // Execute a single call
-    function _execute(Call memory _call) private returns (bool success, bytes memory output) {
-        address to = _call.to == address(0) ? address(this) : _call.to;
-        (success, output) = to.call{value: _call.value}(_call.data);
-    }
-
     /// @dev Keyhash logic not implemented yet
-    function _isValidSignature(bytes32 hash, bytes calldata signature) internal view override returns (bool) {
+    function _isValidSignature(bytes32 hash, bytes calldata signature) internal view override returns (bool isValid) {
         // If the signature's length is 64 or 65, treat it like an secp256k1 signature.
         if (signature.length == 64 || signature.length == 65) {
             return ECDSA.recoverCalldata(hash, signature) == address(this);
         }
         // Otherwise, treat the signature as a wrapped signature, and unwrap it before validating.
-        return _unwrapAndValidate(hash, signature);
+        (isValid,) = _unwrapAndValidateSignature(hash, signature);
     }
 
     /// @dev Returns if the wrapped signature is valid.
     /// TODO: Implement WebAuthnP256 validation.
-    function _unwrapAndValidate(bytes32 digest, bytes calldata wrappedSignature) internal view returns (bool) {
+    function _unwrapAndValidateSignature(bytes32 digest, bytes calldata wrappedSignature)
```

Comment:
this doesn't work if the signature is sepc256k1?

Reply:
ah yeah I think it should actually call _isValidSignature